src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import { expect } from "chai";
+
+import * as projectStatusController from "../../../../server/src/controllers/projectStatusController";
+import * as socket from "../../../src/utils/socket";
+
+
+export function projectStatusControllerTestModule(): void {
+    const projectID = "testProject";
+    it("getAppState of a non-exist project", async() => {
+        const actualResult = await projectStatusController.getAppState(projectID);
+        expect(actualResult).to.equal(projectStatusController.AppState.unknown);
+    });
+
+    it("getBuildState of a non-exist project", async() => {
+        const actualResult = await projectStatusController.getBuildState(projectID);
+        expect(actualResult).to.equal(projectStatusController.BuildState.unknown);
+    });
+
+    it("isBuildRequired of a non-exist project", async() => {
+        const actualResult = await projectStatusController.isBuildRequired(projectID);
+        expect(actualResult).to.equal(false);
+    });
+
+
+
+    it("getAppState of a non-exist project", async() => {
+        const actualResult = await projectStatusController.getAppState(projectID);
+        expect(actualResult).to.equal(projectStatusController.AppState.unknown);
+    });
+
+    it("getBuildState of a non-exist project", async() => {
+        const actualResult = await projectStatusController.getBuildState(projectID);
+        expect(actualResult).to.equal(projectStatusController.BuildState.unknown);
+    });
+
+    it("isBuildRequired of a non-exist project", async() => {
+        const actualResult = await projectStatusController.isBuildRequired(projectID);
+        expect(actualResult).to.equal(false);
+    });
+
+    describe("combinational testing of updateProjectStatus function", () => {
+        before("create workspace settings directory with config file", async () => {
```

Comment:
lets rename the before desc, i think its not correct?

Reply:
Updated